docs/docsite/rst/community/collection_contributors/collection_integration_updating.rst

```diff
@@ -0,0 +1,205 @@
+
+First, we look at the ``tests/integration/targets/<target_name>/tasks/main.yml`` file. In this particular case, the file imports other files from the ``tasks`` directory. The ``postgresql_user_general.yml`` file looks like an appropriate one to add our tests.
+
+.. code-block:: yaml
+
+  # General tests:
+  - import_tasks: postgresql_user_general.yml
+    when: postgres_version_resp.stdout is version('9.4', '>=')
+
+We will add the following code to the file.
+
+.. code-block:: yaml
+
+  # https://github.com/ansible-collections/community.postgresql/issues/NUM
+  # We should also run the same tasks with check_mode: yes. We omit it here for simplicity.
+  - name: Test for new_option, create new user WITHOUT the attribute
+    postgresql_user:
+      name: test_user
+      add_attribute: no
+    register: result
+
+  - name: Check the module returns what we expect
+    assert:
+      that:
+        - result is changed
+
+  - name: Query the database if the user exists but does not have the attribute (it is NULL)
+    postgresql_query:
+      query: SELECT * FROM pg_authid WHERE rolename = 'test_user' AND attribute = NULL
+    register: result
+
+  - name: Check the database returns one row
+    assert:
+      that:
+        - result.query_result.rowcount == 1
+
+  - name: Test for new_option, create new user WITH the attribute
+    postgresql_user:
+      name: test_user
+      add_attribute: yes
+    register: result
+
+  - name: Check the module returns what we expect
+    assert:
+      that:
+        - result is changed
+
+  - name: Query the database if the user has the attribute (it is TRUE)
+    postgresql_query:
+      query: SELECT * FROM pg_authid WHERE rolename = 'test_user' AND attribute = 't'
+    register: result
+
+  - name: Check the database returns one row
+    assert:
+      that:
+        - result.query_result.rowcount == 1
+
+Then we :ref:`run the tests<collection_run_integration_tests>` with ``postgresql_user`` passed as a test target.
+
+In reality, we would alternate the tasks above with the same tasks run with the ``check_mode: yes`` option to be sure our option works as expected in check-mode as well. See :ref:`Recommendations on coverage<collection_integration_recommendations>` for details.
+
+If we expect a task to fail, we use the ``ignore_errors: yes`` option and check that the task actually failed and returned the message we expect:
+
+.. code-block:: yaml
+
+  - name: Test for fail_when_true option
+    postgresql_user:
+      name: test_user
+      fail_when_true: yes
+    register: result
+    ignore_errors: yes
+
+  - name: Check the module fails and returns message we expect
+    assert:
+      that:
+        - result is failed
+        - result.msg == 'The message we expect'
+
+
+.. _collection_run_integration_tests:
+
+Running integration tests
+============================
```

Comment:
There are refs from several docs to this section and it's relevant for the whole topic.
Should we move it to a separate file then and put a link in docs/docsite/rst/community/collection_contributors/collection_integration_tests.rst at the end of the `toctree`?

Reply:
good Idea! I'll push PR shortly that does this.